common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java

```diff
@@ -293,15 +303,46 @@ private Component handleChatLine(
 
     private void handleNpcDialog(List<Component> dialog, NpcDialogueType type) {
         // dialog could be the empty list, this means the last dialog is removed
-        if (!dialog.equals(lastNpcDialog)) {
+        if (dialog.equals(lastNpcDialog)) return;
+
+        if (dialog.isEmpty()) {
+            NpcDialogEvent event = new NpcDialogEvent(dialog, NpcDialogueType.NONE, false);
             lastNpcDialog = dialog;
-            if (dialog.size() > 1) {
-                WynntilsMod.warn("Malformed dialog [#3]: " + dialog);
-                // Keep going anyway and post the first line of the dialog
-            }
-            NpcDialogEvent event = new NpcDialogEvent(dialog, type);
             WynntilsMod.postEvent(event);
+            return;
         }
+
+        if ((System.currentTimeMillis() <= lastSlowdownApplied + SLOWDOWN_PACKET_DIFF_MS)) {
+            // This is a "protected" dialogue if we have gotten slowdown effect just prior to the chat message
+            NpcDialogEvent event = new NpcDialogEvent(dialog, type, true);
+            lastNpcDialog = dialog;
+            WynntilsMod.postEvent(event);
+            return;
+        }
+
+        // Maybe this should be a protected dialogue but packets came in the wrong order.
+        // Wait a tick for slowdown, and then send the event
+        delayedDialogue = dialog;
+        delayedType = type;
+        Managers.MinecraftScheduler.queueRunnable(() -> {
+            if (delayedDialogue != null) {
+                List<Component> dialogToSend = delayedDialogue;
+                delayedDialogue = null;
+                // If we got here, then we did not get the slowdown effect, otherwise we would
+                // have sent the dialogue already
+                NpcDialogEvent event = new NpcDialogEvent(dialogToSend, delayedType, false);
+                lastNpcDialog = dialogToSend;
+                WynntilsMod.postEvent(event);
+                if (type == NpcDialogueType.SELECTION) {
+                    // This is a bit of a workaround to be able to select the options
+                    MutableComponent clickMsg = Component.literal("Open chat and click on the option to select it.")
```

Comment:
Can we? Do we preserve clickables at this point?

Reply:
Yes. (And I have tested this)